packages/core/tests/helpers.ts

```diff
@@ -93,6 +93,7 @@ export async function waitForProofRecord(
     state?: ProofState
     previousState?: ProofState | null
     timeoutMs?: number
+    errorMessage?: string
```

Comment:
You also need to add a `filter` to the `waitForProofRecordSubject` and `waitForCredentialRecordSubject` methods for the `errorMessage` param to be taken into account

Reply:
Sorry @TimoGlastra, I didn't get you.
Can you please explain in detail?

Reply:
Sorry, let me rephrase.

Adding the `errorMessage` to the parameters of `waitForProofRecord` doesn't add `errorMessage` as a filter to the `waitForProofRecord` by itself. So passing `errorMessage` to the `waitForProofRecord` (as you do in the tests) doesn't do anything.

You would need to add an extra line to `waitForProofRecordSubject`:

```ts
.filter((e) => errorMessage === undefined || e.payload.proofRecord.errorMessage === errorMessage),
```

same for credentials